docs/en/stack/ml/nlp/ml-nlp-categorize-text.asciidoc

```diff
@@ -0,0 +1,160 @@
+[[ml-nlp-categorize-text]]
+= Categorize text
+
+:keywords: {ml-init}, {stack}, {nlp}, {lang-ident}, text classification, \
+zero-shot text classification
+
+:description: NLP tasks that classify input text into categories or determine \
+the language of text. 
+
+These NLP tasks enable you to identify the language of text and categorize 
+unstructured input text:
+
+* <<ml-nlp-lang-ident>>
+* <<ml-nlp-text-classification>>
+* <<ml-nlp-zero-shot>>
+
+include::ml-nlp-lang-ident.asciidoc[]
+
+
+[discrete]
+[[ml-nlp-text-classification]]
+== Text classification
+
+Text classification assigns the input text to one of multiple classes or 
+categories that best describe the text. The categories used depend 
+on model training and the dataset used to train with. Based on the
+number of classes, two main types of classification exist: binary
+classification, where the number of classes is exactly two, and
+multi-class classification, where the number of classes is more than
+two.
+
+This task, for example, can help you analyze text for markers of positive or 
+negative sentiment or classify text into various topics.
+
+The example below is of binary classification where the model is trained to 
+perform sentiment analysis with classes "POSITIVE" and "NEGATIVE".
+
+[source,js]
+----------------------------------
+...
+{
+    "input_text": "This was the best movie I’ve seen in the last decade!"
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+
+The task returns the following result:
+
+[source,js]
+----------------------------------
+...
+{
+    "result": "POSITIVE"
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+The second example is of multi-class classification where the model is trained 
+to classify text into news topics with classes "SPORTS", "BUSINESS", "LOCAL", 
+and "ENTERTAINMENT".
+
+[source,js]
+----------------------------------
+...
+{
+    "input_text": "The Blue Jays played their final game in Toronto last night and came out with a win over the Yankees, highlighting just how far the team has come this season."
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+
+The task returns the following result:
+
+[source,js]
+----------------------------------
+...
+{
+    "result": "SPORTS"
+}
+...
+----------------------------------
+// NOTCONSOLE
+
+
+[discrete]
+[[ml-nlp-zero-shot]]
+== Zero-shot text classification
+
+The zero-shot classification task offers the ability to classify text without 
+training a model on a specific set of classes. It uses a model trained on a 
+large dataset that has gained a general language understanding and asks the 
+model how well the labels you provided fit with your text.
+
+This task enables you to analyze and classify your input text even when you 
+don't have sufficient training data to train a text classification model.
+
+Let's see an example:
```

Comment:
Per earlier suggestions, I'd recommend removing the pseudo-results and just summarizing in the introductory text. For example:
```suggestion
For example, you might want to perform multi-class classification and determine whether a news topic is related to "SPORTS", "BUSINESS", "LOCAL", or "ENTERTAINMENT". However, in this case the model is not trained specifically for news categorization; instead, the possible labels are provided together with the input text at inference time:
```